src/main/java/org/variantsync/diffdetective/diff/difftree/serialize/TikzExporter.java

```diff
@@ -0,0 +1,148 @@
+package org.variantsync.diffdetective.diff.difftree.serialize;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.util.Locale;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import org.variantsync.diffdetective.diff.difftree.DiffTree;
+import org.variantsync.diffdetective.util.LaTeX;
+
+/**
+ * Exporter for TikZ pictures which can be embedded into a LaTeX document.
+ *
+ * The resulting graph is styled using TikZ styles which have to be set using {@code \tikzset}. An
+ * example for all required styles can be found in the file {@code tikz_header.tex} in the resource
+ * directory. This particular style is used by {@link exportFullLatexExample}.
+ */
+public final class TikzExporter implements Exporter {
+    private static final Pattern graphvizNodePattern = Pattern.compile("^node (\\w+) ([0-9.]+) ([0-9.]+) .*$");
+    private Format format;
+
+    public TikzExporter(Format format) {
+        this.format = format;
+    }
+
+    /**
+     * Export {@code diffTree} as TikZ graph into {@code destination}.
+     *
+     * The exported graph starts and ends with the {@code tikzpicture} environment and does not
+     * include surrounding LaTeX code like the preamble. For a quick way to set up a working LaTeX
+     * document see {@link exportFullLatexExample}.
+     *
+     * @param diffTree to be exported
+     * @param destination where the result should be written
+     */
+    @Override
+    public void exportDiffTree(DiffTree diffTree, OutputStream destination) throws IOException {
+        exportDiffTree(diffTree, GraphvizExporter.LayoutAlgorithm.DOT, destination);
+    }
+
+    /**
+     * Export {@code diffTree} as TikZ graph into {@code destination}.
+     *
+     * Same as {@link exportDiffTree(DiffTree, OutputStream)}, but allows the selection of
+     * a different Graphviz layout algorithm.
+     */
+    public void exportDiffTree(
+            DiffTree diffTree,
+            GraphvizExporter.LayoutAlgorithm algorithm,
+            OutputStream destination
+            ) throws IOException {
+        // Start tikz picture.
+        var output = new PrintStream(destination);
+        output.println("\\begin{tikzpicture}");
+
+        // Convert the layout information received by Graphviz to coordinates used by TikZ.
+        try (var graphizOutput =
+                new BufferedReader(
+                    new InputStreamReader(
+                        new GraphvizExporter(format)
+                            .computeGraphvizLayout(
+                                diffTree,
```

Comment:
Is this nesting safe? Are all readers closed upon an exception?

Reply:
You're wright, some intermediate constructor might throw an exception (e.g. `OutOfMemoryError`) which would result in possible resource leaks.